.husky/pre-commit

```diff
@@ -0,0 +1,49 @@
+bash << EOF
+  sound() {
+    echo '\a'
+  }
+
+  echo "Running pre-commit hooks..."
+
+# Check Prettier standards
+  npm run format ||
+  (
+      sound
+      echo "❌ Prettier Check Failed. Run npm run format, add changes and try commit again.";
+      exit 1;
+  )
+
+# Check ESLint Standards
+  npm run lint ||
+  (
+      sound
+      echo "❌ ESLint Check Failed. Make the required changes listed above, add changes and try to commit again."
+      exit 1;
+  )
+
+# TODO: add typescript checks
+
+# If everything passes... Now we can commit
+  echo "✅ Checks passed, trying to build..."
+
+  "npm" run build ||
+  (
+      sound
+      echo "❌ Build failed, check errors."
+      exit 1;
+  )
+
+  echo "✅ Successful build, running tests..."
+# After build, run unit tests
+# Right now, runs all tests. Later scope to just unit tets, we can add e2e/integration as github actions on merge
+  npm run test ||
+  (
+      sound
+      echo "❌ Tests failed: View the logs to see what broke and fix it before re-committing."
+      exit 1;
+  )
```

Comment:
We have redundancies here because of the way `pretest` is setup in `package.json`.

Every time we run test, pretest is triggered first which lints, cleans and builds. So in our pre-commit hook runs, the following occurs:
1. format
2. lint
3. build
4. lint
5. clean
6. build
7. test

The only real option is to change pretest so all it does it is clean, otherwise husky will claim tests failed even it was something in the pretest that failed...